pkg/launcher/jvmLauncher.go

```diff
@@ -457,7 +460,94 @@ func (launcher *JvmLauncher) GetRunsByGroup(groupName string) (*galasaapi.TestRu
 // GetRunsById gets the Run information for the run with a specific run identifier
 func (launcher *JvmLauncher) GetRunsById(runId string) (*galasaapi.Run, error) {
 	log.Printf("JvmLauncher: GetRunsById entered. runId=%s", runId)
-	return nil, nil
+
+	var run *galasaapi.Run
+	var err error
+
+	for _, localTest := range launcher.localTests {
+
+		testRunId := localTest.runId
+
+		if testRunId == runId {
+			log.Printf("JvmLauncher: GetRunsById - testRunId '%s' matches with what we're looking for: runId '%s'", testRunId, runId)
+			run, err = createRunFromLocalTest(localTest)
+		}
+	}
+
+	//if we didn't find testRun with same runId log and move on to next run
+	if run == nil {
+		log.Printf("JvmLauncher: GetRunsById - could not find testRun '%s'", runId)
+	} else {
+		log.Printf("JvmLauncher: GetRunsById(testRunId=%s) - testRun returned:%v\n", runId, run.RunId)
+	}
+
+	return run, err
+}
+
+func createRunFromLocalTest(localTest *LocalTest) (*galasaapi.Run, error) {
+
+	var run = galasaapi.NewRun()
+	var err error
+
+	run.SetRunId(localTest.runId)
+
+	if localTest.rasFolderPathUrl == "" {
+		err = fmt.Errorf("createRunFromLocalTest - Don't have enough information to find the structure.json in the RAS folder")
+		log.Printf("%v", err.Error())
+	} else {
+		jsonFilePath := strings.TrimPrefix(localTest.rasFolderPathUrl, "file:///") + "/" + localTest.runId + "/structure.json"
+		log.Printf("createRunFromLocalTest - Reading latest test status from '%s'\n", jsonFilePath)
+
+		err = setTestStructureFromRasFile(run, jsonFilePath, localTest.fileSystem)
+	}
+
+	return run, err
+}
+
+func setTestStructureFromRasFile(run *galasaapi.Run, jsonFilePath string, fileSystem files.FileSystem) error {
+
+	var testStructure = galasaapi.NewTestStructure()
+	var isExists bool
+	var jsonContent string
+
+	url, err := url.Parse(jsonFilePath)
```

Comment:
Why does the json file path have to parse into a URL, which we don't use after that point ?